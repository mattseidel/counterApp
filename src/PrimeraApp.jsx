// import React, { Fragment } from "react";
import React from "react";
import "./index.css";
import PropType from "prop-types";
/**
 * primera app que se realiza
 * @param {Saludo} String un mensaje de saludo
 */
const PrimeraApp = ({ saludo, subtitulo }) => {
  return (
    <>
      <h1>{saludo} </h1>
      <p>{subtitulo}</p>
    </>
  );
  PrimeraApp.protoType = {
    saludo: PropType.string.isRequired,
  };
  PrimeraApp.defaultProps = {
    subtitulo: ""
  }
};

export default PrimeraApp;
