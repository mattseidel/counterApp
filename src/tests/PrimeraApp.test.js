import React from 'react'
import PrimeraApp from "../PrimeraApp"
import { shallow } from 'enzyme'
import '@testing-library/jest-dom'
describe('test on PrimeraApp', () => {
    // test('should show \'hola soy goku\'', () => {
    //     const saludo = 'hola soy Goku'
    //     const { getByText } = render(<PrimeraApp saludo={saludo} />)
    //     expect(getByText(saludo)).toBeInTheDocument();
    // })

    test('should component correctly', () => {
        const saludo = 'hola'
        const wrapper = shallow(<PrimeraApp saludo={saludo} />)
        expect(wrapper).toMatchSnapshot()
    })

    test('should show subtitle sended to props', () => {
        const saludo = 'hola'
        const subtitulo = "soy un subtitulo"
        const wrapper = shallow(
            <PrimeraApp
                saludo={saludo}
                subtitulo={subtitulo}
            />)
        const textoParrafo = wrapper.find('p').text()
        expect(textoParrafo).toBe(subtitulo)

    })


})
