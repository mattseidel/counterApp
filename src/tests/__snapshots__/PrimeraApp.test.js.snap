// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on PrimeraApp should component correctly 1`] = `
<Fragment>
  <h1>
    hola
     
  </h1>
  <p />
</Fragment>
`;
